### ISSUE TITLE
Quagga: Add AS-Path Lists and Route Maps to BGP

### ISSUE THREAD

**Issue by mimugmail** [opnsense/plugins#150]
Hi,

I need some enlightenment! I created two new tabs within BGP. 
First is AS-Path where I can create a ACL style AS Path list.

Here I need to know if theres a form parameter to tell the field to count the number up or forbid using the same ACL number:
![image](https://cloud.githubusercontent.com/assets/20438689/25797808/fe77a5de-33de-11e7-90dc-ccaec9bafdbb.png)

Second one is a bit tricky. In the Route-Map tab I can create multiple route maps and multiple entries for ONE route map and reference an AS-Path list to it:

![image](https://cloud.githubusercontent.com/assets/20438689/25797900/69c990ae-33df-11e7-90bb-b0b5eb42bb62.png)

But this only works when the content of the AS Path list is just text/number etc., but Quagga expects regexp (https://forum.opnsense.org/index.php?topic=5134.0). Any hints here?

Third is most tricky, in the neighbor tab I added the route-map field as a ModelRelationField where I can choose the route map to add. The problem here is, that when I have ONE route map with two entries like:

```
route-map SET_IN_FILTER permit 10
 match as-path 3
 set local-preference 300
!
route-map SET_IN_FILTER permit 20
 set local-preference 290
```

I get two route maps with the same name offered. Is there a form type to tell the UI to only offer one? Or is the logical structure wrong? 

![image](https://cloud.githubusercontent.com/assets/20438689/25798227/9ba7570e-33e0-11e7-9ac2-ee1747b0759c.png)

https://github.com/mimugmail/plugins/blob/bgp/net/quagga/src/opnsense/mvc/app/models/OPNsense/Quagga/BGP.xml#L68

There's also a sub item which I want to add because in the neigbor config I have to choose a direction, but this also doesn't work.

@fichtner I think a list of all available form types and options would be very helpful 👍 

This is my current code base:
https://github.com/mimugmail/plugins/tree/bgp/net/quagga/src/opnsense

Thanks for any help/hint

**Comment by mimugmail** [comment]
I think I've found the list:
https://github.com/opnsense/core/tree/master/src/opnsense/mvc/app/models/OPNsense/Base/FieldTypes

So AutoNumberField should help to create Sequence IDs. 

Perhaps I have to restructure the process of adding route-maps in order to get it done ...

**Comment by mimugmail** [comment]
I think I have fixed most things. Only problem right now is that:

https://docs.google.com/document/d/1h-5y_gEAdfrcJJK3kdCJzlp-zYL1hc8Ejy2hqKpkb4c/edit#heading=h.frna2byt2m84

Here the entry is empty. It doesn't get  the name of the routemap.

Could this probably be that the enum of the routemap is below this block?

In my conf I only get:

```
router bgp 65001
 network 192.168.0.0/24
 neighbor 1.1.1.1 remote-as 65333
 neighbor 1.1.1.1 update-source em0
 neighbor 1.1.1.1 route-map  in
 neighbor 1.1.1.1 route-map  out

```

I tried two differenct approaches for in and out, but nothing workes.

@fabianfrz or @fichtner .. do you have clue?

**Comment by fabianfrz** [comment]
@mimugmail the reason why this may not work is because you are accessing an array field without a key.

**Comment by mimugmail** [comment]
@fabianfrz Thanks! Do you have a reference for a no-brainer like me :)

**Comment by fabianfrz** [comment]
probably this line:
https://github.com/opnsense/plugins/blob/master/net/haproxy/src/opnsense/service/templates/OPNsense/HAProxy/haproxy.conf#L45

**Comment by mimugmail** [comment]
Fixed with #157
